File: ConsoleAppVB1/Demo/DemoAsync.vb
Public Class DemoAsync
    Inherits DemoBase
    Shared Sub Run()
        Dim canExecute As Boolean = True
        Dim op As String = String.Empty


        Console.Clear()
        optionList.Clear()
        optionList.Add("Hello 001...")              '0

        While canExecute
            ShowOptions()
            op = Console.ReadLine()
            Console.WriteLine()
            Console.WriteLine("Your selection is： {0}", op)
            Console.WriteLine()
            Console.WriteLine()
            Select Case op
                Case 0
                    DemoHello.Run()
                Case 1
                    DemoAsync.Run()
                Case "exit"
                    canExecute = False
                Case Else
            End Select
        End While

        Console.WriteLine()
        Console.WriteLine("please type any key !")
        Console.ReadKey()
    End Sub
End Class

File: ConsoleApp1/Demo/DemoAsync.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp1.Demo
{
    class DemoAsync:DemoBase
    {
        public static void Run() {

            bool canExecute = true;
            string op = string.Empty;

            Console.Clear();
            optionList.Clear();
            optionList.Add("Hello");                    //0

            while (canExecute)
            {
                ShowOptions();
                op = Console.ReadLine();
                Console.WriteLine();
                Console.WriteLine("Your selection is： {0}", op);
                Console.WriteLine();
                switch (op)
                {
                    case "0":
                        break;
                    case "exit":
                        canExecute = false;
                        break;
                    default:
                        break;
                }
            }

        }
    }
}
